src/students/students.controller.spec.ts

```diff
@@ -26,7 +19,48 @@ describe('StudentsController', () => {
 
   it('should be defined', () => {
     expect(controller).toBeDefined();
-    controller.findAll(requestMock, statusResponse);
-    expect(statusResponse.status).toHaveBeenCalledWith(200);
+  });
+
+  it('should create new student', async () => {
+    const newStudent = {
+      name: 'newStudent',
+      phone: '0123',
+      address: '123',
+      email: 'newStudent@mail.com',
+      program_id: 1,
+    } as CreateStudentDto;
+
+    const response = await controller.create(newStudent);
+    expect(response.status).toEqual('success');
+    expect(response.data).toBeDefined();
+    expect(response.data).toEqual({ id: expect.any(Number), ...newStudent });
+  });
+
+  it('should find all students', async () => {
+    const response = await controller.findAll();
+    expect(response).toBeDefined();
+    expect(response.length).toBeGreaterThan(0);
+  });
+
+  it('should find one student', async () => {
+    const response = await controller.findOne('1');
+    expect(response).toBeDefined();
+    expect(response.name).toEqual('Student 1');
+  });
+
+  it('should update student', async () => {
+    const updateStudent = {
+      name: 'updatedStudent',
+      phone: '0123',
+      address: '123',
+      email: 'updatedStudent@mail.com',
```

Comment:
Same as I have mentioned. Follow my code. That script is following a specific sequence. It is updating its own inserted data. Try to replace it with Faker or if you want you can five a try with Data Driven Unit Testing with CSV file.

Reply:
nice idea, I am on it